From d990203e82fc9a61e75132ef658f4478fec0ebff Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Fri, 28 Jul 2017 18:05:46 +0800
Subject: [PATCH 2314/5242] MLK-16013-36 staging: typec: wait for cc change on
 ERROR_RECOVERY

commit  b7e8918476d9548ae6eea03893e479f6a2debf51 from
https://source.codeaurora.org/external/imx/linux-imx.git

We should wait at least tErrorRecovery before move to unattached state.

Reviewed-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/typec/tcpm.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/typec/tcpm.c b/drivers/usb/typec/tcpm.c
index 1924d42..4a32b03 100644
--- a/drivers/usb/typec/tcpm.c
+++ b/drivers/usb/typec/tcpm.c
@@ -3701,6 +3701,7 @@ static void _tcpm_cc_change(struct tcpm_port *port, enum typec_cc_status cc1,
 	case PR_SWAP_SRC_SNK_SOURCE_OFF:
 	case PR_SWAP_SRC_SNK_SOURCE_OFF_CC_DEBOUNCED:
 	case PR_SWAP_SNK_SRC_SOURCE_ON:
+	case ERROR_RECOVERY:
 		/*
 		 * CC state change is expected in PR_SWAP
 		 * Ignore it.
-- 
1.7.9.5

